pkg/kubernetes/pod.go

```diff
@@ -140,6 +142,32 @@ func UniquePodName(identifier string) func() string {
 	}
 }
 
+// UniquePodNameWithImage generates a unique pod name that includes both an identifier and an image name.
+// The image name is sanitized to comply with Kubernetes naming conventions.
+func UniquePodNameWithImage(identifier string, imageName string) func() string {
+	return func() string {
+		// Sanitize the identifier
+		identifier = strings.ReplaceAll(identifier, ":", "-")
```

Comment:
I've needed to do sanitization on image name in the past for HTML report generation : https://github.com/radiofrance/dib/blob/main/pkg/report/utils.go#L106

I think we should export this logic to a single location. The `strutil` package seems like a good candidate to me, wdyt ?

Reply:
it seems to be used only to sanitize the pod name